xdsl/dialects/func.py

```diff
@@ -82,6 +91,15 @@ class Return(Operation):
     name: str = "func.return"
     arguments = VarOperandDef(AnyAttr())
 
+    def verify_(self) -> None:
+        func_op: FuncOp = self.parent_op()
```

Comment:
This assumes that the parent_op of the return is a FuncOp. Please assert that this is the case, otherwise we might run into very weird bugs ("dictionary lookup failed" instead of something like "return expects FuncOp as parent").